pom.xml

```diff
@@ -377,6 +385,38 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>scala-2.12</id>
+            <properties>
+                <scala.major.version>2.12</scala.major.version>
+                <scala.version>${scala.major.version}.10</scala.version>
+                <artifact.scala.version>_${scala.major.version}</artifact.scala.version>
+            </properties>
+        </profile>
+        <profile>
+            <id>spark-2.2</id>
```

Comment:
Why do we need to include all these Spark 2.x versions? Spark 2.x versiosn should be compatible, so its fine to compile with one 2.x version and run on a cluster with another version. That should work in the current code base.

Reply:
Wanted to include the other versions in the profile for testing each version. It was mentioned in PR #253 that some more testing should be done. This is building off of that pr and including spark 3 compatibility. If it's overkill we can remove the profiles and keep only 2.4 and 3.0